engine/src/test/java/io/camunda/zeebe/engine/processing/deployment/model/validation/StraightThroughProcessingLoopValidationTest.java

```diff
@@ -0,0 +1,231 @@
+        .contains(String.format("Process: %s", processId))
+        .contains(GENERIC_REJECTION_MESSAGE + "task1 > task2 > task1");
+  }
+
+  @Test
+  public void shouldRejectDeploymentWithComplexStraightThroughProcessingLoop() {
+    // given
+    final var processId = Strings.newRandomValidBpmnId();
+
+    // when
+    // Straight through loop: task1 > parallel2 > manualTask1 > exclusive2 > task1
+    final var rejectedDeployment =
+        ENGINE
+            .deployment()
+            .withXmlResource(
+                Bpmn.createExecutableProcess(processId)
+                    .startEvent()
+                    .userTask("userTask1")
+                    .parallelGateway("parallel1")
+                    .exclusiveGateway("exclusive1")
+                    .task("task1")
+                    .parallelGateway("parallel2")
+                    .manualTask("manualTask1")
+                    .exclusiveGateway("exclusive2")
+                    .defaultFlow()
+                    .userTask("userTask2")
+                    .connectTo("exclusive1")
+                    .moveToNode("parallel2")
+                    .userTask("userTask3")
+                    .endEvent()
+                    .moveToNode("exclusive2")
+                    .sequenceFlowId("sequenceToLoop")
+                    .conditionExpression("true")
+                    .connectTo("task1")
+                    .moveToNode("parallel1")
+                    .task("task2")
+                    .task("task3")
+                    .endEvent()
+                    .done())
+            .expectRejection()
+            .deploy();
+
+    // then
+    Assertions.assertThat(rejectedDeployment)
+        .hasKey(ExecuteCommandResponseDecoder.keyNullValue())
+        .hasRecordType(RecordType.COMMAND_REJECTION)
+        .hasIntent(DeploymentIntent.CREATE)
+        .hasRejectionType(RejectionType.INVALID_ARGUMENT);
+    assertThat(rejectedDeployment.getRejectionReason())
+        .contains(String.format("Process: %s", processId))
+        .contains(
+            GENERIC_REJECTION_MESSAGE
+                + "manualTask1 > exclusive2 > task1 > parallel2 > manualTask1");
+  }
+
+  @Test
+  public void shouldRejectDeploymentWithStraightThroughProcessingLoopNotStartingAtFirstElement() {
+    // given
+    final var processId = Strings.newRandomValidBpmnId();
+
+    // when
+    final var rejectedDeployment =
+        ENGINE
+            .deployment()
+            .withXmlResource(
+                Bpmn.createExecutableProcess(processId)
+                    .startEvent()
+                    .task("task1")
+                    .task("task2")
+                    .task("task3")
+                    .task("task4")
+                    .task("task5")
+                    .task("task6")
+                    .task("task7")
+                    .task("task8")
+                    .task("task9")
+                    .connectTo("task7")
+                    .done())
```

Comment:
:question: :wrench: Can we also have a test with Task - XOR - TASK or did skipped that for now? Not sure anymore.

Reply:
@Zelldon Do you think this is not sufficiently covered with the complex test case? If not I'm happy to add a another case.
<img width="1130" alt="image" src="https://github.com/camunda/zeebe/assets/5787702/a0ae48a0-9ef1-4634-9d51-c0e81ee5de55">

Reply:
Oh sorry @remcowesterhoud I somehow have overseen that.